src/Workspaces/VisualBasic/Portable/Indentation/VisualBasicIndentationService.Indenter.vb

```diff
@@ -26,35 +26,40 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Indentation
 
         Protected Overrides Function GetDesiredIndentationWorker(
                 indenter As Indenter,
-                token As SyntaxToken,
-                previousLine As TextLine,
-                lastNonWhitespacePosition As Integer) As IndentationResult
+                tokenOpt As SyntaxToken?,
+                triviaOpt As SyntaxTrivia?) As IndentationResult
 
-            If token.Span.End = lastNonWhitespacePosition + 1 Then
-                Return GetIndentationBasedOnToken(indenter, token)
-            Else
-                Debug.Assert(token.FullSpan.Contains(lastNonWhitespacePosition))
+            If triviaOpt.HasValue Then
+                Dim trivia = triviaOpt.Value
 
-                Dim trivia = indenter.Root.FindTrivia(lastNonWhitespacePosition)
+                If trivia.Kind = SyntaxKind.CommentTrivia OrElse
+                   trivia.Kind = SyntaxKind.DocumentationCommentTrivia Then
 
-                ' preserve the indentation of the comment trivia before a case statement
-                If trivia.Kind = SyntaxKind.CommentTrivia AndAlso trivia.Token.IsKind(SyntaxKind.CaseKeyword) AndAlso trivia.Token.Parent.IsKind(SyntaxKind.CaseStatement) Then
-                    Return indenter.GetIndentationOfLine(previousLine)
+                    ' if the comment is the only thing on a line, then preserve its indentation for the next line.
+                    Dim line = indenter.Text.Lines.GetLineFromPosition(trivia.FullSpan.Start)
+                    If line.GetFirstNonWhitespacePosition() = trivia.FullSpan.Start Then
+                        Return New IndentationResult(trivia.FullSpan.Start, 0)
+                    End If
                 End If
```

Comment:
this part is similar to C#.  if tehre's a comment standing along on the previous line, we align to it.